packages/styled-components/src/models/StyledComponent.js

```diff
@@ -211,24 +203,24 @@ export default function createStyledComponent(
 
   // fold the underlying StyledComponent attrs up (implicit extend)
   const finalAttrs =
-    // $FlowFixMe
-    isTargetStyledComp && target.attrs
-      ? Array.prototype.concat(target.attrs, attrs).filter(Boolean)
+    isTargetStyledComp && ((target: any): IStyledComponent).attrs
```

Comment:
I would love another option than doing `((target: any): IStyledComponent)` everywhere if you have a suggestion @kitten. Flow doesn't seem to be smart enough to infer things correctly.

Reply:
I'm not sure; you can add a `%checks` directive directly but I think that only works if we remove the intermediate `isTargetStyledComp` variable or if we just make a larger `if` statement branch for when the function returns true